Quantum-Control-Applications/Superconducting/Multi Flux Tunable transmon/Use Case 1 - Two qubit gate optimization with cryoscope/cryoscope.py

```diff
@@ -0,0 +1,224 @@
+                            square_pulse_segments[j].run()
+                # Wait some fixed time so that the whole protocol duration is constant
+                wait(total_len // 4, qbit_name)
+                # Play the second pi/2 pulse along x and y successively
+                with if_(flag):
+                    play("x90", qbit_name)
+                with else_():
+                    play("y90", qbit_name)
+                # State readout
+                align(qbit_name, res_name)
+                measure(
+                    "readout",
+                    res_name,
+                    None,
+                    dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I),
+                    dual_demod.full("rotated_minus_sin", "out1", "rotated_cos", "out2", Q),
+                )
+                save(I, I_st)
+                save(Q, Q_st)
+                # State discrimination
+                if qubit_under_study == 0:
+                    assign(state, I > ge_threshold)
+                else:
+                    assign(state, I > ge_threshold1)
+                save(state, state_st)
+        save(n, n_st)
+
+    with stream_processing():
+        state_st.boolean_to_int().buffer(2).buffer(total_len + 1).average().save("state")
+        n_st.save("iteration")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(host="192.168.88.10", port="80")
+# Open quantum machine
+qm = qmm.open_qm(config)
+# Execute QUA program
+job = qm.execute(cryoscope)
+# Get results from QUA program
+results = fetching_tool(job, data_list=["state", "iteration"], mode="live")
+# Live plotting
+fig = plt.figure()
+interrupt_on_close(fig, job)  # Interrupts the job when closing the figure
+while results.is_processing():
+    # Fetch results
+    state, iteration = results.fetch_all()
+    # Progress bar
+    progress_counter(iteration, n_avg, start_time=results.get_start_time())
+    # Derive results
+    Sxx = state[:, 0] * 2 - 1  # Bloch vector projection along X
+    Syy = state[:, 1] * 2 - 1  # Bloch vector projection along Y
+    S = Sxx + 1j * Syy  # Bloch vector
+    # Qubit phase
+    phase = np.unwrap(np.angle(S))
+    phase = phase - phase[-1]
+    # Qubit detuning
+    detuning = signal.savgol_filter(
+        phase[zeros_before_pulse : const_flux_len - zeros_before_pulse] / 2 / np.pi, 21, 2, deriv=1, delta=0.001
+    )
+    # Step response
+    step_response = detuning / np.average(detuning[-int(const_flux_len / 4)])
+    # plot results
+    plt.subplot(121)
+    plt.cla()
+    plt.plot(xplot, Sxx, ".-", label="Sxx")
+    plt.plot(xplot, Syy, ".-", label="Syy")
+    plt.xlabel("Flux pulse duration [µs]")
+    plt.ylabel("Bloch vector components")
+    plt.title("Cryoscope")
+    plt.legend()
+    plt.subplot(122)
+    plt.cla()
+    plt.plot(xplot[zeros_before_pulse : const_flux_len - zeros_before_pulse], detuning, ".-", label="Pulse")

```

Comment:
correct should be `xplot[zeros_before_pulse : const_flux_len + zeros_before_pulse]`